compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirSupertypesResolution.kt

```diff
@@ -129,8 +132,22 @@ private class FirApplySupertypesTransformer(
             "Expected single supertypeRefs, but found ${supertypeRefs.size} in ${typeAlias.symbol.classId}"
         }
 
+        val supertypeRef = supertypeRefs.single()
+        // TODO: Need better handling for type alias with type parameters.
+        if (typeAlias.typeParameters.isEmpty()) {
+            (supertypeRef.firClassLike(session) as? FirRegularClass)?.let {
+                if (it.classId.isNestedClass) {
+                    val outerClassFir = session.firProvider.getFirClassifierByFqName(it.classId.outerClassId!!) as? FirClass<*>
+                    if (outerClassFir != null) {
+                        (it.scopeProvider.getNestedClassifierScope(outerClassFir, session, scopeSession) as? FirNestedClassifierScope)
```

Comment:
I didn't get why it is necessary but modifying the scope of some class that potentially belongs to an independent module and doesn't know about our type alias looks suspicious

Reply:
Well, that's the point: the scope of a class that doesn't know about the type alias that we're currently resolving here could lead to unresolved constructor call later. I added comments to articulate the situation.